lib/ASRouter.jsm

```diff
@@ -104,17 +114,29 @@ const MessageLoaderUtils = {
         headers.set("If-None-Match", etag);
       }
 
+      let response;
       try {
-        const response = await fetch(provider.url, {headers});
-        if (
-          // Empty response
-          response.status !== 204 &&
-          // Not modified
-          response.status !== 304 &&
-          (response.ok || response.status === 302)
-        ) {
-          remoteMessages = (await response.json())
-            .messages
+        response = await fetch(provider.url, {headers});
+      } catch (e) {
+        MessageLoaderUtils.reportError(e);
+      }
+      if (
+        response &&
```

Comment:
This might be a little too strict, personally I usually go with `status >= 200 && status < 400`